src/kernel/arch/x86/pit.zig

```diff
@@ -149,44 +169,62 @@ inline fn sendCommand(cmd: u8) void {
 }
 
 ///
-/// Read the current reload value of the PIT counter.
+/// Read the current mode of the selected counter.
 ///
 /// Arguments:
-///     counter: u16 - The counter to read from.
+///     IN counter: CounterSelect - The counter to read the mode the counter is operating in.
 ///
-/// Return: u16
-///     The reload value currently in the PIT counter.
+/// Return: u8
+///     The mode the counter is operating in. Use the masks above to get each part.
 ///
-inline fn readCounter(comptime counter: u16) u16 {
-    comptime expect(counter == COUNTER_0_REGISTER or counter == COUNTER_1_REGISTER or counter == COUNTER_2_REGISTER);
-    // Need to disable interrupts as you don't want to set half way through reading
-    arch.disableInterrupts();
-    sendCommand(u8(0));
-
-    const res = u16(arch.inb(counter)) | (u16(arch.inb(counter)) << 8);
-
-    arch.enableInterrupts();
-
-    return res;
+inline fn readBackCommand(counter: CounterSelect) u8 {
+    sendCommand(0xC2);
+    return u8(0x3F) & arch.inb(getRegister(counter));
 }
 
-inline fn read_back_command(comptime counter: u16) u8 {
-    comptime expect(counter == COUNTER_0_REGISTER or counter == COUNTER_1_REGISTER or counter == COUNTER_2_REGISTER);
+///
+/// Send data to a given counter. Will be only one of the 3 counters as is an internal function.
+///
+/// Arguments:
+///     IN counter: CounterSelect - The counter port to send the data to.
+///     IN data: u8               - The data to send.
+///
+inline fn sendDataToCounter(counter: CounterSelect, data: u8) void {
+    arch.outb(getRegister(counter), data);
+}
 
-    sendCommand(u8(0xC2));
-    return u8(0x3F) & arch.inb(counter);
+///
+/// Get the register port for the selected counter.
+///
+/// Arguments:
+///     IN counter: CounterSelect - The counter to get the register port.
+///
+/// Return: u16
+///     The register port for the selected counter.
+///
+fn getRegister(counter: CounterSelect) u16 {
```

Comment:
These functions could be moved into the enum definition, see an example from the zig docs:

```
const Suit = enum {
    Clubs,
    Spades,
    Diamonds,
    Hearts,

    pub fn isClubs(self: Suit) bool {
        return self == Suit.Clubs;
    }
};
```